### PR TITLE
[Optimize] [Gateway] optimize yarn app submit

### PR DESCRIPTION
**Pull request by zackyoungh**
## Verify this pull request

This pull request is code cleanup without any test coverage.

*(or)*

This pull request is already covered by existing tests, such as *(please describe tests)*.

(or)

This change added tests and can be verified as follows:

### CODE REVIEW

**Review by aiwenmo — APPROVED** [review]
LGTM

File: dinky-gateway/src/main/java/org/dinky/gateway/yarn/YarnApplicationGateway.java [hunk 1]
```diff
@@ -77,24 +86,45 @@ public GatewayResult submitJar(FlinkUdfPathContextHolder udfPathContextHolder) {
...
         try (YarnClusterDescriptor yarnClusterDescriptor = createYarnClusterDescriptorWithJar(udfPathContextHolder)) {
             ClusterClientProvider<ApplicationId> clusterClientProvider = yarnClusterDescriptor.deployApplicationCluster(
                     clusterSpecificationBuilder.createClusterSpecification(), applicationConfiguration);
             ClusterClient<ApplicationId> clusterClient = clusterClientProvider.getClusterClient();
-            Collection<JobStatusMessage> jobStatusMessages =
-                    clusterClient.listJobs().get();
 
+            while (yarnClient.getApplicationReport(clusterClient.getClusterId()).getYarnApplicationState()
+                    == YarnApplicationState.ACCEPTED) {
+                Thread.sleep(1000);
```
Comment (leechor) [inline dinky-gateway/src/main/java/org/dinky/gateway/yarn/YarnApplicationGateway.java:97]: Is there a need for a max loop count?

File: dinky-gateway/src/main/java/org/dinky/gateway/yarn/YarnApplicationGateway.java [hunk 2]
```diff
@@ -77,24 +86,45 @@ public GatewayResult submitJar(FlinkUdfPathContextHolder udfPathContextHolder) {
...
+                                yarnClient
+                                                .getApplicationReport(clusterClient.getClusterId())
+                                                .getTrackingUrl() + "jobs/overview"))
+                        .getJSONArray("jobs");
+                jobs.stream().map(x -> (JSONObject) x).forEach(job -> {
+                    JobStatusMessage jobStatusMessage = new JobStatusMessage(
+                            JobID.fromHexString(job.getString("jid")),
+                            job.getString("name"),
+                            JobStatus.valueOf(job.getString("state")),
+                            job.getLong("start-time"));
```
Comment (leechor) [inline dinky-gateway/src/main/java/org/dinky/gateway/yarn/YarnApplicationGateway.java]: in the other Flink version, will "startTime" exist?

**Review by aiwenmo — APPROVED** [review]
LGTM

**Pull request merged**